### ISSUE TITLE
Better documentation of keyword arguments matching: Semgrep matches all arguments in a call when an ellipsis is at the end.

### ISSUE THREAD

**Issue by minusworld** [semgrep/semgrep#1187]
**Describe the bug**
Semgrep matches all arguments in a call when an ellipsis is at the end in Python.

**To Reproduce**
https://semgrep.live/Ew7B

**Expected behavior**
I expected Semgrep to only match line 4, where the supplied keyword argument is at the beginning.

**Comment by disconnect3d** [comment]
**I think there might be another bug then the one described.** I reported this on SemGrep slack and I did not give full description initially. Let me recall the screen I sent there:
![image](https://user-images.githubusercontent.com/10009354/90047001-82442100-dcd1-11ea-81d6-155d68d473c5.png)

I think that this match is correct but the marked description in the learn/tutorial is wrong.

I didn't find it documented anywhere else, but in the semgrep learn/tutorial 5a, we have "Semantics: match keyword arguments in any order" and so it seems to me that if keyword arguments are passed and also the ellipsis operator is used, the order of keyword arguments should be able to mix in with the order of other parameters found with ellipsis operator?

To recall the learn 5a page, we have an example here that shows that semgrep can found results no matter of the order of keyword arguments:
![image](https://user-images.githubusercontent.com/10009354/90047443-35ad1580-dcd2-11ea-8eeb-a3013c475d3d.png)

Now, if we go to the linked reproducer example: https://semgrep.dev/Ew7B, we see that the pattern `requests.get(timeout=30, ...)` matches all results which sounds fine due to the "match keyword arguments in any order" rule. However, if we try the `requests.get(..., timeout=30)` pattern, it does not match the lines where the `timeout=30` keyword argument is not passed as last. **This is a bug imho as it breaks the "match keyword arguments in any order" rule.**

To sum up, it seems there is either a bug regarding to ordering keyword arguments against arguments found via ellipsis operator OR the learn/tutorial pages are wrong and it seems there is not enough documentation around those topics.

**Comment by aryx** [comment]
Yep, keyword arguments are matched in any order, so putting timeout=30 before or after the ellipsis
does not change anything.
